app2/ota_client.py

```diff
@@ -138,9 +139,9 @@ def __init__(self):
 
         self._executor = ThreadPoolExecutor(max_workers=1)
 
-    def update(self, version, url_base, cookies, blocking=False):
+    def update(self, version, url_base, cookies: str, blocking=False):
         try:
-            self._update(version, url_base, cookies, blocking)
+            self._update(version, url_base, json.loads(cookies), blocking)
```

Comment:
`json.loads` will raise `JSONDecodeError` if decoding failed, I think we should handle this?